core/src/test/java/io/questdb/std/DirectByteCharSequenceObjHashMapTest.java

```diff
@@ -138,9 +141,34 @@ public void testHashMapCompatibility() {
                 }
                 p += len;
             }
+        } finally {
+            Unsafe.free(mem, memSize, MemoryTag.NATIVE_DEFAULT);
+        }
+    }
+
+    @Test
+    public void testHashMapUtf8() {
+        final int memSize = 64;
+        long mem = Unsafe.malloc(memSize, MemoryTag.NATIVE_DEFAULT);
+        final DirectByteCharSequence dbcs = new DirectByteCharSequence();
+        DirectByteCharSequenceObjHashMap<Integer> map = new DirectByteCharSequenceObjHashMap<>();
+        try {
+            final String utf16Str = "таблица-табличка";
+            byte[] utf8Bytes = utf16Str.getBytes(StandardCharsets.UTF_8);
+            for (int i = 0; i < utf8Bytes.length; i++) {
+                Unsafe.getUnsafe().putByte(mem + i, utf8Bytes[i]);
+            }
+
+            dbcs.of(mem, mem + utf8Bytes.length);
+
+            Assert.assertNull(map.get(dbcs));
 
+            final StringSink sink = new StringSink();
+            final String utf8Str = sink.put(dbcs).toString();
+            map.put(utf8Str, 42);
+            Assert.assertEquals(42, (int) map.get(dbcs));
```

Comment:
Can you test other methods too, so that putting by CharSequence and getting by string is also same as well as remove etc?

Reply:
Putting by `CharSequence` is not supported by `DirectByteCharSequence*HashMap` collections. Added extra checks and a remove call in 29a795d5361d828080b7a8c2626c6de8a861fce3